File: app/src/main/java/com/android/ashwiask/tvmaze/home/TvMazeApi.java
package com.android.ashwiask.tvmaze.home;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @author Ashwini Kumar.
 */

public interface TvMazeApi {
    @GET("/schedule")
    Single<List<Episode>> getCurrentSchedule(@Query("country") String country,
                                             @Query("date") String date);


    @GET("/shows")
    Single<List<Show>> getShows(@Query("page") int pageNumber);
}

File: app/src/main/java/com/android/ashwiask/tvmaze/network/TvMazeApi.kt
package com.android.ashwiask.tvmaze.network

import com.android.ashwiask.tvmaze.network.home.Episode
import com.android.ashwiask.tvmaze.network.home.Show
import retrofit2.http.GET
import retrofit2.http.Query

interface TvMazeApi {
    @GET("/schedule")
    suspend fun getCurrentSchedule(
        @Query("country") country: String,
        @Query("date") date: String
    ): List<Episode>

    @GET("/shows")
    suspend fun getShows(@Query("page") pageNumber: Int): List<Show>
}
